openedx/core/djangoapps/programs/rest_api/v1/views.py

```diff
@@ -110,39 +105,37 @@ def get(self, request, enterprise_uuid):
 
         return Response(programs)
 
-    def _combine_programs_data_and_progress(self, programs_data, programs_progress):
+    def _combine_programs_data_and_progress(
+        self,
+        programs_data: list[dict | None],
+        programs_progress: list[dict | None],
+    ) -> list[dict | None]:
         """
         Return the combined program and progress data so that api clinet can easily process the data.
         """
         for program_data in programs_data:
             program_progress = next(
-                (
-                    item
-                    for item in programs_progress
-                    if item["uuid"] == program_data["uuid"]
-                ),
+                (item for item in programs_progress if item["uuid"] == program_data["uuid"]),  # type: ignore[index]
```

Comment:
mypy has some limitations on its ability to infer types, and in this case, and a couple of cases I similarly marked ignore,  it just can't figure out what might happen here if the list contains `None`  instead of a dict. It's fine and we don't really care, because `next` and `for item in`  will between them do the right thing, so we just ignore it.